Source/Workers/createVectorTileGeometries.js

```diff
@@ -0,0 +1,346 @@
+
+    function unpackBuffer(buffer) {
+        var packedBuffer = new Float64Array(buffer);
+
+        var offset = 0;
+        Cartesian3.unpack(packedBuffer, offset, scratchCenter);
+        offset += Cartesian3.packedLength;
+
+        Matrix4.unpack(packedBuffer, offset, scratchMatrix4);
+    }
+
+    function packedBatchedIndicesLength(batchedIndices) {
+        var length = batchedIndices.length;
+        var count = 0;
+        for (var i = 0; i < length; ++i) {
+            count += Color.packedLength + 3 + batchedIndices[i].batchIds.length;
+        }
+        return count;
+    }
+
+    function packBuffer(indicesBytesPerElement, batchedIndices, boundingVolumes) {
+        var numBVs = boundingVolumes.length;
+        var length = 1 + 1 + numBVs * BoundingSphere.packedLength + 1 + packedBatchedIndicesLength(batchedIndices);
+
+        var packedBuffer = new Float64Array(length);
+
+        var offset = 0;
+        packedBuffer[offset++] = indicesBytesPerElement;
+        packedBuffer[offset++] = numBVs;
+
+        for (var i = 0; i < numBVs; ++i) {
+            BoundingSphere.pack(boundingVolumes[i], packedBuffer, offset);
+            offset += BoundingSphere.packedLength;
+        }
+
+        var indicesLength = batchedIndices.length;
+        packedBuffer[offset++] = indicesLength;
+
+        for (var j = 0; j < indicesLength; ++j) {
+            var batchedIndex = batchedIndices[j];
+
+            Color.pack(batchedIndex.color, packedBuffer, offset);
+            offset += Color.packedLength;
+
+            packedBuffer[offset++] = batchedIndex.offset;
+            packedBuffer[offset++] = batchedIndex.count;
+
+            var batchIds = batchedIndex.batchIds;
+            var batchIdsLength = batchIds.length;
+            packedBuffer[offset++] = batchIdsLength;
+
+            for (var k = 0; k < batchIdsLength; ++k) {
+                packedBuffer[offset++] = batchIds[k];
+            }
+        }
+
+        return packedBuffer;
+    }
+
+    function createVectorTileGeometries(parameters, transferableObjects) {
+        var boxes = new Float32Array(parameters.boxes);
+        var boxBatchIds = new Uint16Array(parameters.boxBatchIds);
+        var cylinders =  new Float32Array(parameters.cylinders);
+        var cylinderBatchIds = new Uint16Array(parameters.cylinderBatchIds);
+        var ellipsoids =  new Float32Array(parameters.ellipsoids);
+        var ellipsoidBatchIds = new Uint16Array(parameters.ellipsoidBatchIds);
+        var spheres =  new Float32Array(parameters.spheres);
+        var sphereBatchIds = new Uint16Array(parameters.sphereBatchIds);
+
+        var numberOfBoxes = defined(boxes) ? boxBatchIds.length : 0;
+        var numberOfCylinders = defined(cylinders) ? cylinderBatchIds.length : 0;
+        var numberOfEllipsoids = defined(ellipsoids) ? ellipsoidBatchIds.length : 0;
+        var numberOfSpheres = defined(spheres) ? sphereBatchIds.length : 0;
+
+        if (!defined(boxGeometry)) {
```

Comment:
Can `boxGeometry`, etc. be named such that it is more clear that they are file-scoped?

Also, it would be cleaner for each assignment here to have its own `if` statement even though the result is the same.

Reply:
Actually, instead of the file-scoped variables here, should `BoxGeometry, etc. be able to return a default geometry that is lazy computed since this could be useful in many cases?